packages/core/src/types/utils/merkle-tree.interface.ts

```diff
@@ -0,0 +1,56 @@
+import { BigNumber } from '../number'
+
+export interface MerkleTreeNode {
+  key: BigNumber
+  hash: Buffer
+  value: Buffer
+}
+
+export interface MerkleTreeInclusionProof {
+  key: BigNumber
+  value: Buffer
+  siblings: Buffer[]
+}
+
+export interface MerkleTree {
+  /**
+   * Gets the root hash for this tree.
+   *
+   * @returns The root hash.
+   */
+  getRootHash(): Promise<Buffer>
+
+  /**
+   * Updates the provided key in the Merkle Tree to have the value as data,
+   * including all ancestors' hashes that result from this modification.
+   *
+   * @param key The key to update
+   * @param value The new value
+   * @return true if the update succeeded, false if we're missing the intermediate nodes / siblings required for this
+   */
+  update(key: BigNumber, value: Buffer): Promise<boolean>
+
+  /* TODO when we need it
+   * getMerkleProof(key: BigNumber, value: Buffer): Promise<MerkleTreeInclusionProof>
```

Comment:
Just as a note, this will be important to have when integration testing with the contracts, so will be a priority before then.  Let's make a ticket after merging this?